src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -315,7 +316,57 @@ public void parseRequestPathAndQueryTest() throws IOException, URISyntaxExceptio
 			fail(e.getMessage());
 		}
 	}
-	
+
+	@Test
+	public void parseRequestPathWithSpacesAndQueryTest() throws IOException {
+		CompletableFuture<Void> requestReceived = new CompletableFuture<Void>();
+		String extraPath = "/extra Path/extra  Path2";
+		String extraQuery = "queryKey=queryValue&queryKey2=queryValue;"; // mixing in a special char just for fun
+
+		listener.setRequestHandler((context) -> {
+			try {
```

Comment:
Just for completeness, would you mind setting and returning a response inside the handler to make sure that works as well?